src/client/liveblog.ts

```diff
@@ -1,8 +1,38 @@
 // ----- Imports ----- //
 
-import setup from 'client/setup';
 
+import setup from 'client/setup';
+import { fromCapiLiveBlog } from 'item';
+import ReactDOM from 'react-dom';
+import LiveblogBody from 'components/liveblog/body';
+import { createElement as h } from 'react';
+import { Content } from 'mapiThriftModels';
+import { parse } from './parser';
+import { ImageMappings } from 'components/shared/page';
 
 // ----- Run ----- //
 
+declare global {
+    interface Window {
+        __INITIAL__DATA__: Content & {
+            imageMappings: ImageMappings;
+        };
+    }
+}
+
 setup();
+
+const browserParser = (html: string): DocumentFragment =>
+    parse(new DOMParser())(html)
+        .toOption()
+        .withDefault(new DocumentFragment())
```

Comment:
If you want to avoid converting to `Option` this is possible, although may be less clear...

```suggestion
    parse(new DOMParser())(html)
        .either(_ => new DocumentFragment(), identity)
```

Perhaps we should implement `withDefault` for `Result` at some point?